include/pybind11/functional.h

```diff
@@ -58,8 +58,17 @@ struct type_caster<std::function<Return(Args...)>> {
                 if (detail::is_function_record_capsule(c)) {
                     rec = c.get_pointer<function_record>();
                 }
-
                 while (rec != nullptr) {
+                    const int correctingSelfArgument = rec->is_method ? 1 : 0;
+                    if (rec->nargs - correctingSelfArgument != sizeof...(Args)) {
```

Comment:
`rec->nargs != sizeof...(Args) + 1`

to avoid any doubts/tooling noise around unsigned 0 - 1. (probably doesn't matter, but super easy general defensive approach)

Reply:
Now it reads:

```cpp
if (rec->nargs != sizeof...(Args) + self_offset)
```
I assume you indicated like that?